src/numtheory/fp_prime_random_ex.c

```diff
@@ -5,8 +5,10 @@
 /* This is possibly the mother of all prime generation functions, muahahahahaha! */
 int fp_prime_random_ex(fp_int *a, int t, int size, int flags, tfm_prime_callback cb, void *dat)
 {
-   unsigned char *tmp, maskAND, maskOR_msb, maskOR_lsb;
-   int res, err, bsize, maskOR_msb_offset;
+   /* calc the byte size */
+   unsigned char tmp[(size>>3)+(size&7?1:0)];
```

Comment:
```suggestion
   unsigned char tmp[sizeof(a->dp)];
```
Why not simply allocate a statically sized stack array of max size?

This would also require something like the following, after calculating `bsize`.

```
if (bsize > sizeof(tmp))
   return MP_VAL;
```